### ISSUE TITLE
PHP notice relating to is_page usage

### ISSUE THREAD

**Issue by brashrebel** [easydigitaldownloads/EDD-Conditional-Gateways#3]
In v1.0.1 upon activation in WP 4.5.2 I get the following notice:

```
Notice: is_page was called <strong>incorrectly</strong>. Conditional query tags do not work before the query is run. Before then, they always return false. Please see <a href="https://codex.wordpress.org/Debugging_in_WordPress">Debugging in WordPress</a> for more information. (This message was added in version 3.1.) in .../edd.dev/wp-includes/functions.php on line 3897
```

**Comment by brashrebel** [comment]
Also just observed that this only occurs when EDD Recurring Payments is also active.

**Comment by pippinsplugins** [comment]
@ghost1227 Could you look into this?

**Comment by ghost1227** [comment]
@pippinsplugins that error is being thrown by EDD core, not Conditional Gateways... Looks like the `edd_is_checkout` call is throwing it, but I'm not sure why offhand... any ideas?

**Comment by pippinsplugins** [comment]
That usually means you are calling `edd_is_checkout()` too early.

**Comment by pippinsplugins** [comment]
Probably this: https://github.com/easydigitaldownloads/EDD-Conditional-Gateways/blob/207e81d0fa55f3fdb3814244f86475ebf96c1d32/includes/functions.php#L63

Gateways are set up in EDD before the global `$post` object is in WordPress. That's why the error is happening. It'd be best if you didn't use `edd_is_checkout()` to filter the gateways on checkout. You will need to only check `edd_is_checkout()` if `$post` is not empty.

**Comment by ghost1227** [comment]
hmm that check was added because we use the same function to filter on frontend and backend... but it's slightly different. I'll try to write a more stable way of doing it. FYI, the bug didn't happen until the last minor release of Recurring...

**Comment by ghost1227** [comment]
Fixed in 1.0.2
